src/test/java/uk/gov/pay/ledger/event/resource/EventResourceAgreementIT.java

```diff
@@ -85,4 +76,45 @@ public void shouldWriteCancelledEvent() {
         assertThat(agreementEntity.getCancelledByUserEmail(), is("jdoe@example.org"));
         assertThat(agreementEntity.getCancelledDate(), is(ZonedDateTimeMatchers.within(1, ChronoUnit.SECONDS, now)));
     }
+
+    @Test
+    public void shouldNotWriteEventIfAMandatoryFieldIsNotPresent() {
```

Comment:
Just noticed that this is in `EventResourceAgreementIT`. This test class is specifically testing for agreement projection. We can add a test to `EventResourceIT`.

And looks like there is already a test which could be improved https://github.com/alphagov/pay-ledger/blob/10b3c7dfb45475395bd6745dbd514dc8e47008a6/src/test/java/uk/gov/pay/ledger/event/resource/EventResourceIT.java#L62 as doesn't verify status or error message being returned.